packages/content-management/table_list_view_table/src/services.tsx

```diff
@@ -184,6 +187,12 @@ export interface TableListViewKibanaDependencies {
    * Content insights client to enable content insights features.
    */
   contentInsightsClient?: ContentInsightsClientPublic;
+
+  /**
+   * Flag to indicate if Kibana versioning is enabled. (aka not Serverless)
+   * Used to determine if we should show the version mentions in the help text.
+   */
+  isKibanaVersioningEnabled?: boolean;
```

Comment:
I wonder if it wouldn't be best to ask for the `BuildFlavor`? (`'serverless' | 'traditional'`), which is exposed on the `PluginInitializerContext` and internally decide the UI we render based on it. 

We can then pass the `isKibanaVersioningEnabled = buildFlavor === 'traditional'`

This would avoid consumer to pass `true` by mistake when we run in serverless. WDYT?

Reply:
```ts
buildFlavor={serverlessService ? 'serverless' : 'traditional'}
```

Yep that's what I had in mind. The important bit is that we ask the consumer to let us know if we are running in serverless or not. The prop could also be `isServerless`, but then we are limited to a boolean 😊